### DISCUSSION TITLE
I'm trying to use the MediatR in a service Application

### DISCUSSION THREAD

**Discussion by Winterleaf** [LuckyPennySoftware/MediatR#978] [Q&A]
Ok, this is a bit of a complex situation.  I need to host mediatR host inside a Windows service.  I cannot run it in an IIS application due to the complexity of the project, the project has its own built-in system of HTTP Listeners and IIS blocks them if you try to convert the service to a web app.  In the previous versions of MediatR we were able to use your software inside as a service, but it appears the newer Host versions only work inside of a Web Application.  I would really appreciate any help you can give.  I don't know if I'm overlooking something or otherwise.

**Comment by jbogard** [comment]
MediatR can run anywhere `netstandard2.0` can run. There's no dependency on a web host. Any container that can give you an `IServiceProvider` will work fine.


Most people use the MS DI container which works in a ton of hosting environments. Maybe check out [Worker Services](https://learn.microsoft.com/en-us/dotnet/core/extensions/workers)?
